sources/niLang/src/DataTablePath.cpp

```diff
@@ -321,20 +340,19 @@ cString __stdcall cDataTablePath::GetRootPathToDataTable(iDataTable* apDT, tU32
     niCheck(ret != _A("@"),AZEROSTR);
   }
 
-  if  (apDT->GetParent()) {
-    while (apDT->GetParent()) {
-      cString cur;
-      cur = _A("/");
-      cur += apDT->GetName();
-      cur += ret;
-      ret = cur;
-      apDT = apDT->GetParent();
-    }
+  while (apDT->GetParent()) {
+    cString cur;
+    // We use the child index (#N) instead of the name to ensure
+    // the path always leads to the correct property
+    cur << "#";
+    cur << apDT->GetParent()->GetChildIndexFromDataTable(apDT);
+    cur << "/";
+    cur << ret;
```

Comment:
This should be an option as the paths coming out of GetRootPathToDataTable are meant to be readable by a human initially. The index makes it harder to read. That being said if we end up with the XPath syntax which is `ChildName[INDEX]` then I think this becomes moot?

Reply:
Just changed for a somewhat slow algorithm that loops through all the children (until it finds a specific one) to figure out the index of a child for a given name. I'm also dropping the [0] when possible since /Child[0]/@prop equates to /Child/@prop.